Author: Daigo Moriwaki <daigo@debian.org>
Disable gem update --system functionality, as it can break things. Users
should use apt instead. (Closes: #452547).

This patch was imported from the rubygems package.

--- a/lib/rubygems/commands/update_command.rb
+++ b/lib/rubygems/commands/update_command.rb
@@ -128,6 +128,10 @@ class Gem::Commands::UpdateCommand < Gem
   # Update RubyGems software to the latest version.
 
   def update_rubygems
+    if not ENV.include?('REALLY_GEM_UPDATE_SYSTEM')
+      fail "gem update --system is disabled on Debian, because it will overwrite the content of the rubygems Debian package, and might break your Debian system in subtle ways. The Debian-supported way to update rubygems is through apt-get, using Debian official repositories.\nIf you really know what you are doing, you can still update rubygems by setting the REALLY_GEM_UPDATE_SYSTEM environment variable, but please remember that this is completely unsupported by Debian."
+    end
+
     unless options[:args].empty? then
       alert_error "Gem names are not allowed with the --system option"
       terminate_interaction 1
--- a/test/rubygems/test_gem_commands_update_command.rb
+++ b/test/rubygems/test_gem_commands_update_command.rb
@@ -83,7 +83,7 @@ class TestGemCommandsUpdateCommand < Gem
     end
   end
 
-  def test_execute_system
+  def est_execute_system
     util_clear_gems
     util_setup_rubygem9
     util_setup_spec_fetcher @rubygem9
@@ -107,7 +107,7 @@ class TestGemCommandsUpdateCommand < Gem
     assert_empty out
   end
 
-  def test_execute_system_at_latest
+  def est_execute_system_at_latest
     util_clear_gems
     util_setup_rubygem_current
     util_setup_spec_fetcher @rubygem_current
@@ -129,7 +129,7 @@ class TestGemCommandsUpdateCommand < Gem
     assert_empty out
   end
 
-  def test_execute_system_multiple
+  def est_execute_system_multiple
     util_clear_gems
     util_setup_rubygem9
     util_setup_rubygem8
@@ -154,7 +154,7 @@ class TestGemCommandsUpdateCommand < Gem
     assert_empty out
   end
 
-  def test_execute_system_specific
+  def est_execute_system_specific
     util_clear_gems
     util_setup_rubygem9
     util_setup_rubygem8
@@ -179,7 +179,7 @@ class TestGemCommandsUpdateCommand < Gem
     assert_empty out
   end
 
-  def test_execute_system_specifically_to_latest_version
+  def est_execute_system_specifically_to_latest_version
     util_clear_gems
     util_setup_rubygem9
     util_setup_rubygem8
@@ -204,7 +204,7 @@ class TestGemCommandsUpdateCommand < Gem
     assert_empty out
   end
 
-  def test_execute_system_with_gems
+  def est_execute_system_with_gems
     @cmd.options[:args]          = %w[gem]
     @cmd.options[:system]        = true
     @cmd.options[:generate_rdoc] = false
